### DISCUSSION TITLE
Call function on CRUD create

### DISCUSSION THREAD

**Discussion by Read2uk** [filamentphp/filament#14813] [Help] [answered]
### Package

Panel builder

### Package Version

v3

### How can we help you?

Hi,

I am completely new to filament (previously worked with backpack),  thinking of trying it with my new project.

Just a quick question really, is it possible to call a function when creating a CRUD?

For example:
1. I have a CRUD for cars, which includes: reg number, make and model.
2. Add car using CRUD by only adding the reg number.
3. At the point of adding the reg number, make a request api passing the reg number, which then populates the fields with make and model.

Do you have an example how to make such calls on CRUD creation? or posting a job?

Best regards

**Comment by leandrocfe** [comment, marked as answer]
You can [customize the creation process](https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-the-creation-process) or use [hooks](https://filamentphp.com/docs/3.x/panels/resources/creating-records#lifecycle-hooks)

**Reply by Read2uk** [reply]
thank you
